src/mjolnir/ingest_transit.cc

```diff
@@ -257,18 +257,29 @@ void setup_stops(Transit& tile,
   auto onestop_id = isGenerated ? tile_stop.stop_id + "_" + to_string(node_type) : tile_stop.stop_id;
   node->set_onestop_id(onestop_id);
   if (node_type == NodeType::kMultiUseTransitPlatform) {
-    stop_graphIds[{onestop_id, feed_path}] = node_id;
+    // when this platform is not generated, we can use its actual given id verbatim because thats how
+    // other gtfs entities will refer to it. however when it is generated we must use its parents id
+    // and not the generated one because the references in the feed have no idea that we are doing
+    // the generation of new ideas for non-existant platforms
+    platform_node_ids[{tile_stop.stop_id, feed_path}] = node_id;
   }
   node_id++;
 }
 
+/**
```

Comment:
from here down there  are pretty much no real important changes, its all me re-naming the generic "stop_graphids" to the specific "platform_node_ids". i think this variable name makes it much more clear what information is going in there even though for the generated ones the gtfs ids will be the parent station (but it is the one to use to get that platforms graphid)